### PR TITLE
Do not allow invalid op names in TFUtilities::isStatefulOp

### PR DESCRIPTION
**Pull request by bgogul**
Resolves [SR-9388](https://bugs.swift.org/browse/SR-9388).

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFUtilities.cpp [hunk 1]
```diff
@@ -264,20 +264,25 @@ bool tf::isStatefulOp(const GraphOperationInst *graphOp) {
...
   // Is this a stateful TensorFlow op?
   std::unique_ptr<TF_Status, decltype(&TF_DeleteStatus)> status(
       TF_NewStatus(), TF_DeleteStatus);
   int isStateful = TF_OpIsStateful(tensorOpName.str().c_str(), status.get());
   TF_Code statusCode = TF_GetCode(status.get());
-  // FIXME(SR-9388): we should actually fail if statusCode != TF_OK. However,
-  // the current tests use arbitrary strings in graph_op. Therefore, we will
-  // just allow TF_NOT_FOUND for now to avoid changing too many tests.
-  assert((statusCode == TF_OK || statusCode == TF_NOT_FOUND) &&
+  assert(statusCode == TF_OK &&
```
Comment (mhong) [inline lib/SILOptimizer/Mandatory/TFUtilities.cpp]: would this assert trigger, when user writes `#tfop()` directly, with a bad op name?

we might need this function to return a `status` for error instead.

**Review by mhong — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/TFUtilities.cpp (hunk 1 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: Yes. Don't we need to prevent someone from using an invalid op name?

Comment (mhong) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: It shouldn't check fail, but should report error diagnostics.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow asan

File: lib/SILOptimizer/Mandatory/TFUtilities.cpp (hunk 1 above)
Comment (bgogul) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: I see what you are saying. If we want to report error diagnostics, wouldn't it make sense to make name validation as a part of GraphOperationInst construction itself?

Comment (mhong) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: Agree that checking name earlier would be more preferred. Would you like to file a bug to track that?

Comment (bgogul) [reply lib/SILOptimizer/Mandatory/TFUtilities.cpp]: Done. https://bugs.swift.org/browse/SR-9499

Will submit this PR then.

**Pull request merged**